test/client.test.js

```diff
@@ -71,6 +71,65 @@ test('keys() returns an array', (t) => {
     .catch(t.threw)
 })
 
+test('scan() returns an array when only cursor is provided', (t) => {
+  t.plan(4)
+  const client = factory({ client: new MockIoredis() })
+  client.set('foo', 'foo', 1000)
+    .then(() => client.scan(0))
+    .then((result) => {
+      t.type(result, Array)
+      t.type(result[0], 'number')
+      t.type(result[1], Array)
+      t.is(result[1][0], 'foo')
+    })
+    .catch(t.threw)
+})
+
+test('scan() returns an array when match pattern is provided', (t) => {
+  t.plan(4)
+  const client = factory({ client: new MockIoredis() })
+  client.set('foo', 'foo', 1000)
+    .then(() => client.scan(0, 'MATCH', 'fo*'))
+    .then((result) => {
+      t.type(result, Array)
+      t.type(result[0], 'number')
+      t.type(result[1], Array)
+      t.is(result[1][0], 'foo')
+    })
+    .catch(t.threw)
+})
+
+test('scan() returns an array when match pattern and count is provided', (t) => {
+  t.plan(5)
+  const client = factory({ client: new MockIoredis() })
+  client.set('foo', 'foo', 1000)
+    .then(() => client.set('bar', 'bar', 1000).then(
+      () => client.scan(0, 'MATCH', '*', 'COUNT', 10))
+      .then((result) => {
+        t.type(result, Array)
+        t.type(result[0], 'number')
+        t.type(result[1], Array)
+        t.is(result[1][0], 'foo')
+        t.is(result[1][1], 'bar')
+      })
+    )
+    .catch(t.threw)
+})
+
+test('scanStream() returns an readable stream', (t) => {
+  t.plan(2)
+  const client = factory({ client: new MockIoredis() })
```

Comment:
Does `MockIoredis` truly support this method? The `disconnect` and `quit` tests actually hit a local Redis server in a Docker container. Should this test do the same?

Reply:
Yes, it does support scanStream. Tests are passing with that as well.

Reply:
I don't see it listed in https://github.com/stipsan/ioredis-mock/blob/ade6c8caad7f5aeb1deb5fe3cc55e0bfa52cfa08/compat.md . Is that a mistake in their documenation?

Reply:
Probably the docs is only showing "redis" commands, not scanStream method. I don't see any stream related methods there..

<img width="759" alt="Screenshot 2021-04-13 at 10 25 56 AM" src="https://user-images.githubusercontent.com/22556323/114577950-179d1800-9c9a-11eb-8f8d-c41d1525040e.png">

I tried on runkit 🔝 , the repo is using different version. But It still has the method.

Reply:
Btw, while testing this out just now, I noticed `Readable` is supported from node 10.17 onwards. So this `ScanStream` method only works for latter version of node, I hope that's okay, as older versions are deprecated anyway.